bench/src/main/scala/parsley/bench/JsonBench.scala

```diff
@@ -0,0 +1,51 @@
+package parsley.bench
+
+import java.util.concurrent.TimeUnit
+import org.openjdk.jmh.annotations._
+import org.typelevel.jawn.ast._
+import parsley._, Combinator._, Parsley._
+import scala.io.Source
+
+/* Based on https://github.com/typelevel/jawn/blob/v1.0.0/benchmark/src/main/scala/jawn/JmhBenchmarks.scala */
+@State(Scope.Benchmark)
+@BenchmarkMode(Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+abstract class JmhBenchmarks(name: String) {
+  val text: String =
+    Source.fromResource(name, getClass.getClassLoader).getLines().mkString("\n")
+
+  def json: Parsley[JValue] = {
+    val jsontoks = LanguageDef("", "", "", false, NotRequired, NotRequired, NotRequired, NotRequired, Set.empty, Set.empty, true, Predicate(Char.isWhitespace))
+    val tok = new TokenParser(jsontoks)
+    lazy val obj: Parsley[JValue] = tok.braces(tok.commaSep(+(tok.stringLiteral <~> tok.colon *> value)).map(pairs => JObject.fromSeq(pairs)))
+    lazy val array: Parsley[JValue] = tok.brackets(tok.commaSep(value)).map(list => JArray.fromSeq(list))
+    lazy val value: Parsley[JValue] =
+      (tok.stringLiteral.map(JString.apply)
+        <|> tok.symbol("true") *> Parsley.pure(JTrue)
+        <|> tok.symbol("false") *> Parsley.pure(JFalse)
+        <|> tok.symbol("null") *> Parsley.pure(JNull)
+        <|> array
+        <|> attempt(tok.float).map(JNum.apply)
+        <|> tok.integer.map(i => JNum.apply(i.toLong))
+        <|> obj)
+
+    tok.whiteSpace *> (obj <|> array) <* eof
+  }
+
+  @Benchmark
+  def parsleyParseCold(): JValue =
+    runParserThreadSafe(json, text).toOption.get
+
+  // Stable instance to warm up
+  val hotJson: Parsley[JValue] = json
```

Comment:
`hotJson` isn't going to work as you expect by the way, you need to use `.force()` to get it to properly compile the parser in advance, the generation of the instructions is lazy.